### PR TITLE
Add function of estimating step

### PR DESCRIPTION
**Pull request by boyeon555**
- Add debug dispatcher
- Used for accessing to citizen node especially now for an API of estimating step
- Add a validator for estimating step without stepLimit and signature field

### CODE REVIEW

File: iconrpcserver/server/json_rpc/dispatcher/__init__.py [hunk 1]
```diff
@@ -16,3 +16,4 @@
 from .version2 import *
```
Comment (whylee259) [inline iconrpcserver/server/json_rpc/dispatcher/__init__.py]:
```suggestion
```

Comment (boyeon555) [reply iconrpcserver/server/json_rpc/dispatcher/__init__.py]: Applied it.

**Review by cow-hs — APPROVED** [review]
good job

**Review by whylee259 — APPROVED** [review]

**Review by inwonkim — APPROVED** [review]

File: docs/icon-json-rpc-v3.md [hunk 2]
```diff
@@ -1070,6 +1086,67 @@ It is used when transfering a message, and `data` has a HEX string.
 }
 ```
 
+## debug_estimateStep
+
+* Generates and returns an estimates step cost of how much step is necessary to allow the transaction to complete as not being added to the blockchain. Note that the estimate may be more than the actual amount of a step cost used by the transaction for several reasons including node performance.
```
Comment (sink772) [inline docs/icon-json-rpc-v3.md]: I think it's better to split the first sentence into two like below. And use just step, not step cost.
```suggestion
* Generates and returns an estimated step of how much step is necessary to allow the transaction to complete. The transaction will not be added to the blockchain. Note that the estimate may be more than the actual amount of step used by the transaction for several reasons including node performance.
```

File: docs/icon-json-rpc-v3.md [hunk 3]
```diff
@@ -1070,6 +1086,67 @@ It is used when transfering a message, and `data` has a HEX string.
...
+| value     | [T_INT](#T_INT)                                            | optional | Amount of ICX coins in loop to transfer. When ommitted, assumes 0. (1 icx = 1 ^ 18 loop) |
+| timestamp | [T_INT](#T_INT)                                            | required | Transaction creation time. timestamp is in microsecond.      |
+| nid       | [T_INT](#T_INT)                                            | required | Network ID ("0x1" for Mainnet, "0x2" for Testnet, etc)       |
+| nonce     | [T_INT](#T_INT)                                            | optional | An arbitrary number used to prevent transaction hash collision. |
+| dataType  | [T_DATA_TYPE](#T_DATA_TYPE)                                | optional | Type of data. (call, deploy, or message)                     |
+| data      | T_DICT or String                                           | optional | The content of data varies depending on the dataType. See [Parameters - data](#sendtxparameterdata). |
+
+### Returns
+
+* The amount of an estimated step cost in loop
```
Comment (sink772) [inline docs/icon-json-rpc-v3.md]:
```suggestion
* The amount of an estimated step in loop
```

File: docs/icon-json-rpc-v3.md [hunk 4]
```diff
@@ -1070,6 +1086,67 @@ It is used when transfering a message, and `data` has a HEX string.
 }
 ```
 
+## debug_estimateStep
+
+* Generates and returns an estimates step cost of how much step is necessary to allow the transaction to complete as not being added to the blockchain. Note that the estimate may be more than the actual amount of a step cost used by the transaction for several reasons including node performance.
+
+### Parameters
+
+* The transaction's information not to have stepLimit and signature keys
```
Comment (sink772) [inline docs/icon-json-rpc-v3.md]:
```suggestion
* The transaction information not to have stepLimit and signature
```

**Pull request merged**